Feature/login social

taejunUM opened the pull request:
## 📄 소셜 로그인 & 회원가입 (카카오, 네이버, 구글) 구현
<!-- ex. feat: 회원가입 API 구현 -->

#### 📎 관련 이슈
<!-- 연결된 이슈 번호 -->
close #5 

## ✅ 변경 사항
<!-- 어떤 작업을 했는지 간략히 정리 -->
- 소셜 로그인, 회원가입 기능 구현 [Kakao, Naver, Google]
- 클라이언트로부터 전달받은 SocialAccessToken 서버 전달, 서버에서 직접 검증 및 사용자 정보 요청
- TokenVerifier 구현 소셜 플랫폼별 사용자 정보 요청 로직 분리 구현
- SocialAccessToken (소셜 플랫폼 accessToken) 인증 성공시 JWT 발급
    - 기존 사용자: 인증 성공시(200 OK) JWT 발급
    - 신규 사용자: 임시 JWT 발급(401 Unauthorized), 추가 정보 작성 완료 후 로그인 및 JWT 재발급 
- AbstractTokenVerifier 공통 서명/검증 로직 최소화
- JWT 재발급
    - ClubMemberRole 변경시 JWT 재발급
    - 기존 JWT 클라이언트에서 삭제

## 🔍 테스트 방법
<!-- 어떤 방식으로 동작을 검증했는지 -->
- [x] 로컬에서 직접 테스트
- [ ] 유닛 테스트 작성 및 통과
- [ ] 브라우저 / 앱 화면에서 확인

## 👀 중점적으로 봐줬으면 하는 부분
<!-- 리뷰어가 집중해서 봐야 할 부분이 있다면 -->
- JwtFilter 내 토큰 검증 로직이 적절한지
- TokenVerifier 구조의 확장성

## 📝 기타 사항
<!-- 추가적으로 공유할 내용 -->
- 현재 구조에서는 Refresh Token 미도입 상태로, JWT 무효화 시 자동 재발급 불가, 로직상에서 재발급 (클라이언트에서 기존 JWT 삭제 필요)
- 추후 Refresh Token 및 토큰 재발급 API 도입을 고려

Hojin02 committed:
feat: ClubMember를 찾는 메소드 추가

taejunUM committed:
refactor: build.gradle security 주석 해제

taejunUM committed:
feat: 기본 securityConfig 구현

taejunUM committed:
style: 주석 추가

taejunUM committed:
feat: JwtFilter 구현

taejunUM committed:
refactor: build.gradle.kts security 주석 제거

- 기존 build.gradle 삭제

taejunUM committed:
refactor: SecurityConfig filter 추가

taejunUM committed:
refactor: Security Config 인가 경로 수정

taejunUM committed:
feat: CustomUserDetails 구현

taejunUM committed:
feat: CustomUserDetailsService 구현

taejunUM committed:
refactor: CustomUserDetailsService 패키지 이동

taejunUM committed:
refactor: build.gradle Oauth2 주석 해제

taejunUM committed:
refactor: Member 엔티티 password 필드 삭제

taejunUM committed:
refactor: 클래스 삭제

taejunUM committed:
refactor: password 제거

taejunUM committed:
refactor: 메서드 수정

taejunUM committed:
refactor: socialProvider, socialId 주석 제거

taejunUM committed:
refactor: SecurityConfig oauth2 추가

taejunUM committed:
refactor: SecurityConfig import추가

taejunUM committed:
refactor: MemberValidator 메소드 추가

taejunUM committed:
refactor: JwtFilter 메소드 수정

taejunUM committed:
feat: OAuth2UserInfo 인터페이스 구현

taejunUM committed:
feat: OAuth2UserInfo 구현체 생성

- Kakao, Naver, Google

taejunUM committed:
feat: OAuth2 성공 핸들러 구현

taejunUM committed:
feat: CustomOAuth2UserService 구현

taejunUM committed:
feat: SocialProvider 구현

taejunUM committed:
refactor: CustomUserDetails 리팩토링

taejunUM committed:
refactor: CustomUserDetailsService 리팩토링

taejunUM committed:
refactor: SecurityConfig 내부 JwtFilter 파라미터 추가

taejunUM committed:
refactor: 어노테이션 추가

taejunUM committed:
refactor: attributes socialId 추가

taejunUM committed:
feat: .yml oauth2 설정 추가

taejunUM committed:
refactor: 파라미터 네이밍 변경

- provider > socialProvider

taejunUM committed:
refactor: ddl-auto 설정 변경

taejunUM committed:
refactor: 패키지 이동

taejunUM committed:
refactor: SecurityConfig 설정 변경

taejunUM committed:
refactor: dto 네이밍 변경 적용

taejunUM committed:
feat: 상세정보 입력 폼 구현

taejunUM committed:
refactor: kakao 테스트앱 설정

taejunUM committed:
refactor: kakaoDto getName 파라미터 수정

- nickname > name

taejunUM committed:
refactor: 재가입 예외 수정

- CustomUserDetailsService > AuthService

taejunUM committed:
refactor: SuccessHandler 토큰 발급 로직 수정

- 보안 설정 추가
- 쿠키 설정

taejunUM committed:
refactor: 미사용 메소드 정리

- MemberRepository
- MemberValidator

taejunUM committed:
feat: AuthController 구현

- 회원가입&로그인 컨트롤러

taejunUM committed:
feat: AuthDetailsInfoController 구현

- 회원가입 추가 정보 입력 폼 컨트롤러

taejunUM committed:
feat: AuthService 구현

- 회원가입 서비스

taejunUM committed:
feat: 가입 성공 리다이렉트 html

- 편의성 고려 개발 환경 임시 html
- 토큰 반환, 복사 기능

taejunUM committed:
reafctor: build.gradle thymeleaf 설정 추가

- 소셜로그인, 로그인 성공 html 사용

taejunUM committed:
refactor: 메소드 네이밍, 상태코드 변경

taejunUM committed:
refactor: Token 쿠키 설정 방식 ResponseCookie 변경

- 기존 Cookie는 SameSite 지원X ResponseCookie로 교체

taejunUM committed:
feat: 로그인 성공 Redirect용 html 작성

- 개발 단계에서만 유지

taejunUM committed:
style: 공백 제거

taejunUM committed:
refactor: Valid 어노테이션 추가

taejunUM committed:
style: 공백 제거

taejunUM committed:
style: 주석 내용 변경

Hojin02 committed:
refactor: Article 상세 조회 로직변경에 따른 필드 및 메소드 수정

taejunUM committed:
style: 공백 제거

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/controller/AuthController.java:
```diff
@@ -1,28 +1,31 @@
     ) {
-        return ResponseEntity.status(HttpStatus.OK)
-                .body(authService.login(dto));
+        authService.registerNewUser(dto);
```
현재 user 대신 member 로 회원 도메인을 지칭하고 있습니다 네이밍 변경해주세요

taejunUM replied:
수정완

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,50 @@
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+        }
+        filterChain.doFilter(request, response);
```
if 문 종료 후에는 한칸의 공백을 띄워주세요

taejunUM replied:
수정 완료

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,50 @@
+                        userDetails, null, userDetails.getAuthorities());
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+        }
```
if 문이 중첩되어 있습니다 중첩하지 않고 처리할 수 있는 방법을 고민해보는게 좋을 것 같아요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,50 @@
+        String bearerToken = request.getHeader(AUTHORIZATION);
+        if (bearerToken != null && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(BEARER.length());
+        }
```
if 문 시작과 끝은 한칸씩 공백을 주는게 좋아보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/oauth2/service/CustomOAuth2UserService.java:
```diff
@@ -0,0 +1,63 @@
+    public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
+        OAuth2User oAuth2User = delegate.loadUser(userRequest);
+
+
```
공백은 한칸만 띄워주세요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/oauth2/service/CustomOAuth2UserService.java:
```diff
@@ -0,0 +1,63 @@
+public class CustomOAuth2UserService extends DefaultOAuth2UserService {
+
+    private final MemberValidator memberValidator;
+    private final OAuth2UserService<OAuth2UserRequest, OAuth2User> delegate = new DefaultOAuth2UserService();
```
DefaultOAuth2UserService 를 상속받고 있는데 DefaultOAuth2UserService 인스턴스를 delegate 로 가지고 있을 필요는 없을 것 같습니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/oauth2/service/CustomOAuth2UserService.java:
```diff
@@ -0,0 +1,63 @@
+                    );
+                    return new CustomUserDetails(null, true, newAttributes);
+                });
+    }
```
return 키워드는 한칸씩 띄워주세요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/oauth2/service/CustomOAuth2UserService.java:
```diff
@@ -0,0 +1,63 @@
+                    );
+                    return new CustomUserDetails(null, true, newAttributes);
+                });
+    }
```
new 키워드 대신 정적 팩토리 메서드로 생성해주세요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/user/service/CustomUserDetailsService.java:
```diff
@@ -0,0 +1,36 @@
+            throw new UsernameNotFoundException(e.getMessage());
+        }
+    }
+}
```
DomainException 으로 감싸주는게 좋을 것 같습니다

taejunUM replied:
수정 완료

taejunUM committed:
refactor: 메소드 네이밍 변경

taejunUM committed:
style: 공백 추가

taejunUM committed:
refactor: 중첩 if문 분리

taejunUM committed:
style: 공백 제거

taejunUM committed:
refactor: 상속받은 기능 중복 선언 필드 제거

- delegate 필드 삭제 및 super.로 대체

taejunUM committed:
style: 공백 추가

taejunUM committed:
refactor: new > 정적 팩토리 메서드

taejunUM committed:
refactor: null 반환 > 빈 리스트 반환

- NPE 우려

taejunUM committed:
refactor: 예외처리 수정

- DomainException으로 감싸서 처리

taejunUM committed:
refactor: 메서드 네이밍 오타 수정

taejunUM committed:
style: 주석 추가 (Todo)

taejunUM committed:
refactor: 컨트롤러 입력방식 어노테이션 변경

taejunUM committed:
refactor: 파라미터 네이밍 변경

taejunUM committed:
refactor: 메서드 통합

- 중복 쿼리 방지

taejunUM committed:
feat: Login 요청 dto 구현

taejunUM committed:
feat: Login 응답 dto

taejunUM committed:
refactor: dto 네이밍 수정

taejunUM committed:
style: 들여쓰기

taejunUM committed:
refactor: 미사용 컨트롤러, HTML 제거

taejunUM committed:
refactor: 컨트롤러 로직 수정

taejunUM committed:
refactor: 로그인 성공 dto 수정

taejunUM committed:
refactor: 소셜 로그인 서비스 로직 수정

taejunUM committed:
refactor: 소셜로그인 담당 컨트롤러, 서비스 네이밍 변경

- AuthLogin > SocialLogin

taejunUM committed:
refactor: CustomOAuth2UserService 로직 수정

taejunUM committed:
refactor: 소셜 로그인 서비스 로직 수정

taejunUM committed:
feat: 소셜토큰 검증 인터페이스 구현

taejunUM committed:
feat: google 소셜 토큰 검증 구현체

- google 소셜 accessToken 검증 & 사용자 정보 요청

taejunUM committed:
feat: kakao 소셜 토큰 검증 구현체

- kakao 소셜 accessToken 검증 & 사용자 정보 요청

taejunUM committed:
feat: naver 소셜 토큰 검증 구현체

- naver 소셜 accessToken 검증 & 사용자 정보 요청

taejunUM committed:
feat: TokenVerifier 분기처리 로직 구현

taejunUM committed:
feat: SocialUserInfo 구현

- 각 소셜에서 받은 사용자 정보 dto로 반환

taejunUM committed:
refactor: google 요청 정보 네이밍 변경

- sub > id

taejunUM committed:
refactor: 예외처리 강화, 유효성 검증 책임 분리

taejunUM committed:
refactor: 미사용 클래스 제거

- SecurityConfog oauth2Login 미사용으로 연관 클래스 삭제

taejunUM committed:
refactor: 미사용 dto 제거

- 토큰은 쿠키에 저장되므로 JSON 반환 필요X

taejunUM committed:
refactor: 로그인 & 회원가입 Controller 리팩토링

- 토큰 반환 방식 변경
- JSON > 쿠키 저장

taejunUM committed:
refactor: 회원가입 & 로그인 Service 리팩토링

taejunUM committed:
feat: CookieUtil 구현

- JWT 토큰 쿠키에 저장
- 토큰 기본 설정

taejunUM committed:
refactor: JwtFilter 리팩토링

- oauth2Login > 커스텀 방식으로 변경에 따른 리팩토링

taejunUM committed:
refactor: SecurityConfig 리팩토링

- oauth2Login > 커스텀 방식으로 변경에 따른 리팩토링

taejunUM committed:
refactor: MemberRepository 쿼리메서드 수정

taejunUM committed:
refactor: MemberValidator 메서드 리팩토링

taejunUM committed:
refactor: member생성 API 삭제

taejunUM committed:
refactor: member 생성 메서드 삭제

taejunUM committed:
feat: 소셜 플랫폼용 RestClient 설정 추가

- 소셜 사용자 정보 요청을 위한 RestClient Bean 등록

taejunUM committed:
feat: 소셜로그인 에러코드 정의

taejunUM committed:
refactor: google id 변수명 수정

- id > sub

taejunUM committed:
refactor: google 사용자 정보 요청 경로 변경

- v2 > v3

taejunUM committed:
feat: WebConfig 구현

- CORS 설정 추가

taejunUM committed:
refactor: 사용자 정보 요청 API 구현

- 플랫폼마다 CORS 제약이 다르므로 백엔드에서 직접 요청하는 API

taejunUM committed:
refactor: HTML 제거

taejunUM committed:
refactor: 미사용 클래스 제거

taejunUM committed:
refactor: 로그인 API 리팩토링

- 기존, 신규 동일하게 토큰 발급

taejunUM committed:
refactor: 로그인 Service 리팩토링

- 기존 멤버 JWT 발급
- 신규 유저 임시용 JWT 발급

taejunUM committed:
feat: 추가정보 입력 API 구현

taejunUM committed:
feat: 신규유저용 추가정보 입력, 회원가입 서비스 로직 구현

taejunUM committed:
feat: SocialUserInfo 반환 dto

taejunUM committed:
feat: SocialLoginResponseDto 구현

- 토큰 반환, 기존&신규 멤버 구별

taejunUM committed:
refactor: 추가정보 요청 dto 리팩토링

- 토큰에서 데이터 반환 후 사용

taejunUM committed:
refactor: 토큰 유효시간 개별 설정

taejunUM committed:
feat: 회원가입용 임시 토큰 발급 메서드 구현

taejunUM committed:
feat: 임시토큰에서 사용자 정보 반환 메서드 구현

taejunUM committed:
refactor: 멤버 검증 메서드 리팩토링

taejunUM committed:
feat: TokenVerifier 추상클래스 도입

taejunUM committed:
refactor: 추상클래스 구현체 리팩토링

- 추상메서드 @Override

taejunUM committed:
refactor: account 패키지 리팩토링

taejunUM committed:
refactor: 미사용 메서드 제거

# 본문은 아래에 작성

# 꼬릿말은 아래에 작성: ex) Github issue #23

# --- COMMIT END ---
# feat	    새로운 기능 추가
# fix	    버그 수정
# docs	    문서 변경 (README 등 문서 파일 수정)
# style	    코드 스타일 변경 (세미콜론, 들여쓰기, 공백 등)
# refactor  코드 리팩터링 (기능 변화 없음, 변수명 정리 등)
# test	    테스트 코드 추가 및 수정
# chore	    빌드 설정, 패키지 매니저 설정, 기타 중요하지 않은 작업
# ------------------
#   커밋 메시지는 영문 소문자로 시작
#   형식은 type: message 사용
#   하나의 커밋에는 하나의 목적만 작성
#   메시지는 명확하게 무엇을 했는지 설명
# ------------------
#   <Optional>
#   하위 커밋 설명이 필요한 경우, 본문(body)을 줄바꿈하여 추가 설명 작성
#   feat: support social login
#
#   Added Kakao and Google login support.
#   Refactored auth controller to handle provider-based login.

taejunUM committed:
style: 줄바꿈 수정

taejunUM committed:
refactor: socialProvider 리팩토링

- 미사용 메서드 정리
- fromString 메서드 @JsonCreator 추가

taejunUM committed:
refactor: verifier 패키지 리팩토링

taejunUM committed:
style: 줄바꿈

taejunUM committed:
style: 주석 추가

taejunUM committed:
refactor: 어노테이션 추가

- @Transactional

taejunUM committed:
refactor: comment 테스트 코드 수정

- Member.createMember 메서드 파라미터 수정

taejunUM committed:
refactor: import 추가

- merge 과정에서 import 안 된 것 추가

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/controller/SignUpDetailsInfoController.java:
```diff
@@ -0,0 +1,38 @@
+
+    @GetMapping("/details-info")
+    public ResponseEntity<SignUpDetailsInfoResponseDto> getSocialUserInfo(
+            @CookieValue("accessToken") String signupToken
```
accesstoken 의 경우 Authorization header 로 받는게 더 안전할 것 같아요. 제 3자 토큰을 브라우저 또는 쿠키에 남아있는건 좋지 않아 보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/controller/SignUpDetailsInfoController.java:
```diff
@@ -0,0 +1,38 @@
+
+    @PostMapping("/sign-up")
+    public ResponseEntity<Void> signup(
+            @CookieValue("accessToken") String signupToken,
```
위와 같은 이유로 쿠키 사용보다는 header 를 사용하는게 좋아보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/controller/SocialLoginController.java:
```diff
@@ -0,0 +1,35 @@
+            @RequestBody @Valid SocialLoginRequestDto dto
+    ) {
+        SocialLoginResponseDto result = socialLoginService.socialLogin(dto);
+        ResponseCookie cookie = CookieUtil.createAccessTokenCookie(result.accessToken());
```
여기서 반환하고 있는 토큰은 서버의 jwt 토큰으로 보이는데 소셜 서비스의 accesstoken 과 구분하기 어려운 네이밍을 가지고 있는것 같습니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/GoogleUserInfoResponse.java:
```diff
@@ -0,0 +1,49 @@
+    public String getName() {
+        if (attributes.get("name") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/GoogleUserInfoResponse.java:
```diff
@@ -0,0 +1,49 @@
+        this.attributes = attributes;
+        if (attributes == null || attributes.get("sub") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
if 문과 주입 로직의 순서가 변경되는게 조금 더 효율적이지 않을까요?
그리고 통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/GoogleUserInfoResponse.java:
```diff
@@ -0,0 +1,49 @@
+    public String getEmail() {
+        if (attributes.get("email") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/KakaoUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+        this.attributes = attributes;
+        if (attributes == null || attributes.get("id") == null || attributes.get("kakao_account") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
if 문과 주입 로직의 순서가 변경되는게 조금 더 효율적이지 않을까요?
그리고 통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/KakaoUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+        Map<String, Object> kakaoAccount = (Map<String, Object>) attributes.get("kakao_account");
+        if (kakaoAccount == null || kakaoAccount.get("name") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/KakaoUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+        Map<String, Object> kakaoAccount = (Map<String, Object>) attributes.get("kakao_account");
+        if (kakaoAccount == null || kakaoAccount.get("email") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/NaverUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+
+        if (this.response == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/NaverUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+    public String getProviderId() {
+        if (response.get("id") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/NaverUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+    public String getName() {
+        if (response.get("name") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/dto/response/NaverUserInfoResponse.java:
```diff
@@ -0,0 +1,51 @@
+    public String getEmail() {
+        if (response.get("email") == null) {
+            throw new DomainException(OAuth2ErrorCode.NO_USER_INFO);
+        }
```
통일성을 위해 중괄호 없이 if 문을 작성하는게 좋아보입니다

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/oauth/verifier/AbstractSocialTokenVerifier.java:
```diff
@@ -0,0 +1,39 @@
+        } catch (Exception e) {
+            throw new DomainException(OAuth2ErrorCode.FAILED_TO_REQUEST_USER_INFO);
+        }
+        return parseUserInfo(response);
```
return 키워드는 한칸의 공백을 가지는게 컨벤션입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/service/SignUpDetailsInfoService.java:
```diff
@@ -0,0 +1,35 @@
+        Member member = dto.toEntity(userInfo);
+        memberRepository.save(member);
+
+        return tokenProvider.createAccessToken(member.getId());
```
소셜 서비스의 accesstoken 과 우리 서버의 jwt 토큰의 이름은 다르게 하는게 좋을것같아요

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/auth/util/CookieUtil.java:
```diff
@@ -0,0 +1,18 @@
+                .maxAge(Duration.ofDays(1))    // 쿠키 유효시간
+                .build();
+    }
+}
```
토큰을 header 로 전달하면 이 util 클래스는 필요하지 않을것같습니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/account/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,62 @@
+            Long memberId = tokenProvider.getAccessTokenInfo(token);
+
+            try {
+                Member member = memberValidator.findMemberByMemberIdOrThrow(memberId);
```
jwt 를 사용하는 이유는 근본적으로 무상태를 유지하기 위함입니다. 토큰을 가지고 요청을 할때마다 db 에 조회를 하는건 좋지 않아 보입니다

taejunUM replied:
수정 완료

Developer-Groo commented on src/main/java/com/mobble/mobbleserver/config/WebConfig.java:
```diff
@@ -0,0 +1,19 @@
+                .allowedHeaders("*")
+                .exposedHeaders("Set-Cookie");            // ✅ 쿠키 응답 헤더 노출
+    }
+}
```
현재 서비스는 앱에서의 요청만 고려하고 있기 때문에 cors 설정은 필요하지 않습니다

taejunUM replied:
수정 완료

taejunUM committed:
refactor: 토큰 네이밍 변경

taejunUM committed:
style: 공백 추가

# 본문은 아래에 작성

# 꼬릿말은 아래에 작성: ex) Github issue #23

# --- COMMIT END ---
# feat	    새로운 기능 추가
# fix	    버그 수정
# docs	    문서 변경 (README 등 문서 파일 수정)
# style	    코드 스타일 변경 (세미콜론, 들여쓰기, 공백 등)
# refactor  코드 리팩터링 (기능 변화 없음, 변수명 정리 등)
# test	    테스트 코드 추가 및 수정
# chore	    빌드 설정, 패키지 매니저 설정, 기타 중요하지 않은 작업
# ------------------
#   커밋 메시지는 영문 소문자로 시작
#   형식은 type: message 사용
#   하나의 커밋에는 하나의 목적만 작성
#   메시지는 명확하게 무엇을 했는지 설명
# ------------------
#   <Optional>
#   하위 커밋 설명이 필요한 경우, 본문(body)을 줄바꿈하여 추가 설명 작성
#   feat: support social login
#
#   Added Kakao and Google login support.
#   Refactored auth controller to handle provider-based login.

taejunUM committed:
refactor: 토큰 생성 메서드 네이밍 변경

- accessToken > accessJwtToken

taejunUM committed:
reafctor: 토큰 저장방식 변경

- 쿠키 > 헤더

taejunUM committed:
refactor: CookieUtil 제거

- 쿠키 미사용 유틸클래스 제거

taejunUM committed:
refactor: WebConfig 제거

- CORS 미사용 클래스 제거

taejunUM committed:
refactor: if문 컨벤션 통일

taejunUM committed:
refactor: if문 컨벤션 통일

taejunUM committed:
refactor: if문 컨벤션 통일

taejunUM committed:
refactor: 파라미터 네이밍 변경

taejunUM committed:
refactor: 토큰 파라미터 네이밍 변경

taejunUM committed:
refactor: 토큰 파라미터 네이밍 변경

taejunUM committed:
style: 주석 수정

taejunUM committed:
fix: signup 토큰 생성 로직 순서 변경

- setClame() -> setSubject() 순서 변경
- subject먼저 설정하면 subject가 제거됨

taejunUM committed:
refactor: JwtFilter 검증 로직 수정

- 요청시마다 DB조회 > DB조회 부분 제거
- CustomUserDetails 대신 memberId로 사용하여 @AuthenticationPrincipal 사용시 바로 memberId 추출 가능

taejunUM committed:
refactor: JwtFilter 생성자 파라미터 제거

taejunUM committed:
style: 들여쓰기 변경

taejunUM committed:
Merge branch 'develop' into feature/login-social

taejunUM committed:
refactor: CustomUserDetails 제거

- Spring Security oauth2Login() 구조를 사용하지 않는 상태에선 CustomUserDetails는 필요하지 않음

taejunUM committed:
feat: AuthMember 클래스 추가

- memberId 직접 주입 대신 객체로 래핑해서 주입

taejunUM committed:
refactor: JwtFilter 로직 수정

- memberId > 객체로 감싸서 주입

taejunUM committed:
refactor: TokenProvider 리팩토링

taejunUM committed:
refactor: JwtFilter 리팩토링

- Role 권한 추가

taejunUM committed:
feat: ClubMemberRole List로 가져오는 메서드 추가

taejunUM committed:
refactor: 권한 조회, 토큰 생성시 전달 로직 추가

taejunUM committed:
style: 주석 수정

taejunUM committed:
refactor: 파라미터 네이밍 변경

taejunUM committed:
refactor: tokenVersion 컬럼 추가

taejunUM committed:
refactor: tokenVersion 증가 메서드 작성

taejunUM committed:
refactor: TokenProvider 리팩토링

- JWT 생성 시 tokenVersion 추가
- JWT 에서 tokenVersion 추출 메서드 작성
- tokenVersion 추출 시 타입 검증, 예외처리

taejunUM committed:
refactor: JwtFilter 리팩토링

- JWT tokenVersion, DB tokenVersion 비교 검증
- JWT tokenVersion이 이전 버전일 때 인증 실패 > 재로그인 흐름

taejunUM committed:
refactor: clubMember 권한 변경 메소드 리팩토링

- 권한 변경 시 member.tokenVersion 증가

taejunUM committed:
style: 구현해야할 흐름 Todo 작성

taejunUM committed:
refactor: 클래스 네이밍 변경

- Oauth2 > OAuth

taejunUM committed:
refactor: oauth 에러코드 정의

taejunUM committed:
refactor: 예외처리 DomainException 변경

taejunUM committed:
refactor: 신규 유저 HTTP 상태코드 변경

taejunUM committed:
feat: logout API 구현

taejunUM committed:
refactor: 로그인 api url 변경

taejunUM committed:
refactor: Security JwtFilter memberValidator 주입

taejunUM committed:
refactor: oauth 에러메시지 추가

taejunUM committed:
refactor: Signup Token 검증 제외

- subject 값이 memberId(Long) 값이 아닌 경우 예외
- JwtFilter 검증시 memberId(Long) 값이 아닌 경우 토큰 검증 제외

taejunUM committed:
Merge branch 'develop' into feature/login-social

taejunUM committed:
Merge branch 'develop' into feature/login-social

taejunUM committed:
refactor: tokenVersion 필드 제거

taejunUM committed:
refactor: 로그아웃 api 제거

- tokenVersion 제거로 로그아웃은 클라이언트 역할로 위임

taejunUM committed:
refactor: TokenProvider 토큰 버전 제거

taejunUM committed:
refactor: tokenVersion 관련 ErrorCode 제거

taejunUM committed:
refactor: JwtFilter tokenVersion 검증 제거

taejunUM committed:
style: 공백 제거

taejunUM committed:
refactor: 불필요한 주입 제거

taejunUM committed:
feat: 기존 응답 래핑용 dto 구현

taejunUM committed:
refactor: 클럽 권한 수정시 JWT 재발행

taejunUM committed:
refactor: 클럽 권한 변경 API 반환타입 수정

- 래핑용 DTO로 변환

taejunUM committed:
refactor: 불필요한 memberValidator 주입 제거

taejunUM committed:
style: 줄바꿈

taejunUM committed:
refactor: 토큰 네이밍 변경

- 명확한 네이밍 구별
- accessJwtToken > jwtToken

taejunUM committed:
refactor: JWT Token 생성 메서드 네이밍 변경

taejunUM committed:
refactor: 토큰 네이밍 변경

taejunUM committed:
style: 주석 수정

taejunUM committed:
refactor: JwtFilter.doFilterInternal 로직 수정

taejunUM committed:
refactor: 토큰 네이밍 변경

taejunUM committed:
style: 주석 추가

taejunUM committed:
Merge branch 'develop' into feature/login-social

# Conflicts:
#	build.gradle.kts
#	src/main/java/com/mobble/mobbleserver/domain/clubMember/service/ClubMemberService.java
